pkg/providers/aws/credentials.go

```diff
@@ -136,14 +137,24 @@ type Credentials struct {
 	PolicyName      string
 	AccessKeyID     string
 	SecretAccessKey string
+	RoleArn         string
+	TokenFilePath   string
 }
 
 //go:generate moq -out credentials_moq.go . CredentialManager
 type CredentialManager interface {
 	ReconcileProviderCredentials(ctx context.Context, ns string) (*Credentials, error)
-	ReconcileSESCredentials(ctx context.Context, name, ns string) (*Credentials, error)
-	ReoncileBucketOwnerCredentials(ctx context.Context, name, ns, bucket string) (*Credentials, *v1.CredentialsRequest, error)
-	ReconcileCredentials(ctx context.Context, name string, ns string, entries []v1.StatementEntry) (*v1.CredentialsRequest, *Credentials, error)
+	ReconcileBucketOwnerCredentials(ctx context.Context, name, ns, bucket string) (*Credentials, error)
+}
+
+func NewCredentialManager(client client.Client) CredentialManager {
+	ns, _ := k8sutil.GetOperatorNamespace()
+	_, err := getSTSCredentialsSecret(context.TODO(), client, ns)
+	if err == nil {
```

Comment:
Could we be more specific around the error handling here. I think the case is that If the error is isNotFound we use credentialsminter but if it is found we use sts?

Reply:
Yes, that makes sense.
But to properly handle other errors we would have to return the err from this function and handle it in the layers above. And I suspect that it might be anti-pattern to return errors from this "NewSomething" constructor-like function, so I'll have to look into this and maybe come up with different approach.